core/src/test/java/com/gentics/mesh/linkrenderer/BranchLinkRendererTest.java

```diff
@@ -112,19 +121,53 @@ public void setupDeps() {
 		nodeCreateRequest.setFields(fields);
 		localNodeUuid = call(() -> client().createNode(projectName(), nodeCreateRequest, new VersioningParametersImpl().setBranch(branchName)))
 			.getUuid();
+
+		updatedUuid = updateExistingNode();
+	}
+
+	/**
+	 * Update existing node in default branch only
+	 * @return
+	 */
+	private String updateExistingNode() {
+		NodeResponse node = call(() -> client().webroot(PROJECT_NAME, "/Deals")).getNodeResponse();
+		NodeUpdateRequest updateRequest = node.toRequest();
+		node.getFields().putString("slug", "changedDeals");
+		call(() -> client().updateNode(PROJECT_NAME, node.getUuid(), updateRequest));
+
+		return node.getUuid();
 	}
 
 	@Test
 	public void testResolve() {
-		String prefix = getPrefix();
-		try (Tx tx = tx()) {
+		String replacedContent = replaceContent("{{mesh.link('" + contentUuid + "')}}");
+		assertEquals("Check rendered content", getPrefix() + "/News/News%20Overview.en.html", replacedContent);
+	}
+
+	@Test
+	public void testResolveWithBranchParameter() {
+		String replacedContent;
+
+		// New Branch:
+
+		replacedContent = replaceContent(String.format("{{mesh.link('%s', 'en', '%s')}}", updatedUuid, branchUuid));
```

Comment:
Please also add testcases for error behavior. What happens when branchUuid is bogus or branchUuid is e.g. a userUuid.